### ISSUE TITLE
systemd user service

### ISSUE THREAD

**Issue by clarfon** [mozilla/sccache#555]
It'd be nice if a systemd user service were provided so that systemd could manage starting the server. That way, I could set the server to start when I log in so that I don't have to wait for it to start on the first invocation, which has lead to timeouts in some cases.

Additionally, it'd help logging so that I can see when the server starts/stops and if it has any errors.

**Comment by emilio** [comment]
I got it to work like this: #556 

But not sure whether that's the best way to do that. Providing it by default somehow would be great of course.

**Comment by clarfon** [comment]
Honestly, this is what I would use:

```
[Unit]
Description=Start sccache server
Wants=network.target
After=network.target

[Service]
ExecStartPre=/usr/bin/sccache --stop-server
ExecStart=/usr/bin/sccache --start-server
ExecStop=/usr/bin/sccache --stop-server
Type=forking

[Install]
WantedBy=default.target
```

Mostly, I think it'd be good to include in the official distributions.

**Comment by emilio** [comment]
Ah, but that's for the client/daemon, not for builders, it's two different things I guess :)

**Comment by clarfon** [comment]
Yeah, that's what I meant. A global server would be run as a system service.

**Comment by Stargateur** [comment]
If like me you install sccache as user you can use `%h/.cargo/bin/sccache`.
